In [1]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [1]:
import pickle
import time
import cknn
import torch
import gc

import numpy as np
import pandas as pd
import DataFormatterEditAddSession as DataFormatterA
import DataFormatterEdit as DataFormatter

from model import *
from tqdm.notebook import tqdm
from utils import build_graph, split_validation, Data

In [2]:
#import importlib
#importlib.reload()
#from IPython.display import display

In [3]:
def num_data(data, str_):
    values = data.loc[:,[str_]].values
    values = values.reshape(-1,)
    new = np.unique(values)
    return new.shape[0]

In [4]:
fold = 5
#_path = 'amz_data_prep_2018_book.csv'
_path = 'amz_data_prep_2018_book_40k.csv'

In [5]:
def make_pd_data_frame(train_data, session_key='session_id', item_key='item_id', time_key='eventdate'):
    df = pd.DataFrame({session_key:[], item_key:[], time_key:[]})
    ses_ids = []
    ses_dates = []
    ses_seqs = []
    for ses_id, ses_date, ses_seq in tqdm(zip(train_data[0], train_data[1], train_data[2]), total=len(train_data[0])):
        for ses_ in ses_seq:
            ses_date_ = int(ses_date.astype(np.float))
            ses_ids.append(int(ses_id))
            ses_dates.append(ses_date_)
            ses_seqs.append(ses_)
    ses_ids = np.asarray(ses_ids)
    ses_dates = np.asarray(ses_dates)
    ses_seqs = np.asarray(ses_seqs)
    d = np.concatenate((ses_ids,ses_seqs,ses_dates))
    d = np.reshape(d,(3,-1))
    d = d.T
    column_names = [session_key, item_key, time_key]
    df = pd.DataFrame(d, columns=column_names)
    return df

In [6]:
data = pd.read_csv(_path, delimiter=',')
n_item = num_data(data, 'item_id')
k_split_data_list, sess_clicks = DataFormatterA.get_k(_path, fold)
last_sr_gnn_hit, sr_gnn_hit, iknn_hit, final_hit, last_sr_gnn_mrr, sr_gnn_mrr, iknn_mrr, final_mrr = [], [], [], [], [], [], [], []
for __p in range(5):
    avg_p = []
    avg_m = []
    for fold_ in range(fold):
        gc.collect()
        test_data_preproc = k_split_data_list[fold_]
        train_data_preproc = k_split_data_list.copy()
        del train_data_preproc[fold_]
        train_data_ = _ = train_data = test_data = df = knn = mrr_20 = p_20 = scr = scr_convd = slices = None

        train_data_preproc, test_data_preproc, train_data_, _ = DataFormatterA.make_tr_ts_data_set(train_data_preproc, 
                                                                                                      test_data_preproc, 
                                                                                                      sess_clicks)

        train_data = Data(train_data_preproc, shuffle=True)
        test_data = Data(test_data_preproc, shuffle=False)
        #test_data.inputs.shape
        #train_data_preproc = [[subset],[ans_items]]
        #train_data_ = [[session_id],[session_date],[item_seq]]
        #train_data = obj
        n_node = n_item+1

        model = trans_to_cuda(SessionGraph(n_node))

        best_result = [0, 0]
        best_epoch = [0, 0]
        bad_counter = 0

        for epoch in range(20):
            hit, mrr = train_test(model, train_data, test_data)
            flag = 0
            if hit > best_result[0]:
                best_result[0] = hit
                best_epoch[0] = epoch
                flag = 1
            if mrr > best_result[1]:
                best_result[1] = mrr
                best_epoch[1] = epoch
                flag = 1
            bad_counter += 1 - flag
            if bad_counter >= 5:
                break
        print('-------------------------------------------------------')
        last_sr_gnn_hit.append(hit)
        last_sr_gnn_mrr.append(mrr)

        df = make_pd_data_frame(train_data_)
        knn = cknn.ContextKNN(data = df, session_key='session_id', item_key='item_id', time_key='eventdate')

        mrr_20, p_20, scr = knn.multi_predict_ts_data(test_data, 12)


        model.eval()
        hit, mrr = [], []
        slices = test_data.generate_batch(model.batch_size)

        for i in tqdm(slices):
            targets, scores = forward(model, i, test_data)
            scores = trans_to_cpu(scores).detach().numpy()
            for j, idx in enumerate(i):
                v = scr[idx][0]
                for index, val in zip(list(v.keys()), list(v.values())):
                    scores[j,index-1] += val
            scores = torch.Tensor(scores)
            sub_scores = scores.topk(20)[1]
            sub_scores = trans_to_cpu(sub_scores).detach().numpy()
            for score, target, mask in zip(sub_scores, targets, test_data.mask):
                hit.append(np.isin(target - 1, score))
                if len(np.where(score == target - 1)[0]) == 0:
                    mrr.append(0)
                else:
                    mrr.append(1 / (np.where(score == target - 1)[0][0] + 1))

        final_hit.append(np.mean(hit)*100)
        final_mrr.append(np.mean(mrr)*100)

        sr_gnn_hit.append(best_result[0])
        sr_gnn_mrr.append(best_result[1])

        iknn_hit.append(p_20)
        iknn_mrr.append(mrr_20)
    print()
    print('SR-GNN Score: ', np.mean(last_sr_gnn_hit), np.mean(last_sr_gnn_mrr))
    print('SR-GNN Last Score: ', np.mean(sr_gnn_hit), np.mean(sr_gnn_mrr))
    print('CKNN Score: ', np.mean(iknn_hit), np.mean(iknn_mrr))
    print('Proposed Method Score: ',np.mean(final_hit), np.mean(final_mrr))
    avg_p.append(np.mean(final_hit))
    avg_m.append(np.mean(final_mrr))
print('Method 1 Avg Proposed Method Score: ',np.mean(avg_p), np.mean(avg_m))
gc.collect()
for __p in range(5):
    avg_p = []
    avg_m = []
    for fold_ in range(fold):
        gc.collect()
        test_data_preproc = k_split_data_list[fold_]
        train_data_preproc = k_split_data_list.copy()
        del train_data_preproc[fold_]
        train_data_ = _ = train_data = test_data = df = knn = mrr_20 = p_20 = scr = scr_convd = slices = None

        train_data_preproc, test_data_preproc, train_data_, _ = DataFormatterA.make_tr_ts_data_set(train_data_preproc, 
                                                                                                      test_data_preproc, 
                                                                                                      sess_clicks)

        train_data = Data(train_data_preproc, shuffle=True, method=0)
        test_data = Data(test_data_preproc, shuffle=False)
        #test_data.inputs.shape
        #train_data_preproc = [[subset],[ans_items]]
        #train_data_ = [[session_id],[session_date],[item_seq]]
        #train_data = obj
        n_node = n_item+1

        model = trans_to_cuda(SessionGraph(n_node))

        best_result = [0, 0]
        best_epoch = [0, 0]
        bad_counter = 0

        for epoch in range(20):
            hit, mrr = train_test(model, train_data, test_data)
            flag = 0
            if hit > best_result[0]:
                best_result[0] = hit
                best_epoch[0] = epoch
                flag = 1
            if mrr > best_result[1]:
                best_result[1] = mrr
                best_epoch[1] = epoch
                flag = 1
            bad_counter += 1 - flag
            if bad_counter >= 5:
                break
        print('-------------------------------------------------------')
        last_sr_gnn_hit.append(hit)
        last_sr_gnn_mrr.append(mrr)

        df = make_pd_data_frame(train_data_)
        knn = cknn.ContextKNN(data = df, session_key='session_id', item_key='item_id', time_key='eventdate')

        mrr_20, p_20, scr = knn.multi_predict_ts_data(test_data, 12)


        model.eval()
        hit, mrr = [], []
        slices = test_data.generate_batch(model.batch_size)

        for i in tqdm(slices):
            targets, scores = forward(model, i, test_data)
            scores = trans_to_cpu(scores).detach().numpy()
            for j, idx in enumerate(i):
                v = scr[idx][0]
                for index, val in zip(list(v.keys()), list(v.values())):
                    scores[j,index-1] += val
            scores = torch.Tensor(scores)
            sub_scores = scores.topk(20)[1]
            sub_scores = trans_to_cpu(sub_scores).detach().numpy()
            for score, target, mask in zip(sub_scores, targets, test_data.mask):
                hit.append(np.isin(target - 1, score))
                if len(np.where(score == target - 1)[0]) == 0:
                    mrr.append(0)
                else:
                    mrr.append(1 / (np.where(score == target - 1)[0][0] + 1))

        final_hit.append(np.mean(hit)*100)
        final_mrr.append(np.mean(mrr)*100)

        sr_gnn_hit.append(best_result[0])
        sr_gnn_mrr.append(best_result[1])

        iknn_hit.append(p_20)
        iknn_mrr.append(mrr_20)
    print()
    print('SR-GNN Score: ', np.mean(last_sr_gnn_hit), np.mean(last_sr_gnn_mrr))
    print('SR-GNN Last Score: ', np.mean(sr_gnn_hit), np.mean(sr_gnn_mrr))
    print('CKNN Score: ', np.mean(iknn_hit), np.mean(iknn_mrr))
    print('Proposed Method Score: ',np.mean(final_hit), np.mean(final_mrr))
    avg_p.append(np.mean(final_hit))
    avg_m.append(np.mean(final_mrr))
print('Method 2 Avg Proposed Method Score: ',np.mean(avg_p), np.mean(avg_m))
gc.collect()
for __p in range(5):
    avg_p = []
    avg_m = []
    for fold_ in range(fold):
        gc.collect()
        test_data_preproc = k_split_data_list[fold_]
        train_data_preproc = k_split_data_list.copy()
        del train_data_preproc[fold_]
        train_data_ = _ = train_data = test_data = df = knn = mrr_20 = p_20 = scr = scr_convd = slices = None

        train_data_preproc, test_data_preproc, train_data_, _ = DataFormatterA.make_tr_ts_data_set(train_data_preproc, 
                                                                                                      test_data_preproc, 
                                                                                                      sess_clicks)

        train_data = Data(train_data_preproc, shuffle=True, method=1)
        test_data = Data(test_data_preproc, shuffle=False)
        #test_data.inputs.shape
        #train_data_preproc = [[subset],[ans_items]]
        #train_data_ = [[session_id],[session_date],[item_seq]]
        #train_data = obj
        n_node = n_item+1

        model = trans_to_cuda(SessionGraph(n_node))

        best_result = [0, 0]
        best_epoch = [0, 0]
        bad_counter = 0

        for epoch in range(20):
            hit, mrr = train_test(model, train_data, test_data)
            flag = 0
            if hit > best_result[0]:
                best_result[0] = hit
                best_epoch[0] = epoch
                flag = 1
            if mrr > best_result[1]:
                best_result[1] = mrr
                best_epoch[1] = epoch
                flag = 1
            bad_counter += 1 - flag
            if bad_counter >= 5:
                break
        print('-------------------------------------------------------')
        last_sr_gnn_hit.append(hit)
        last_sr_gnn_mrr.append(mrr)

        df = make_pd_data_frame(train_data_)
        knn = cknn.ContextKNN(data = df, session_key='session_id', item_key='item_id', time_key='eventdate')

        mrr_20, p_20, scr = knn.multi_predict_ts_data(test_data, 12)


        model.eval()
        hit, mrr = [], []
        slices = test_data.generate_batch(model.batch_size)

        for i in tqdm(slices):
            targets, scores = forward(model, i, test_data)
            scores = trans_to_cpu(scores).detach().numpy()
            for j, idx in enumerate(i):
                v = scr[idx][0]
                for index, val in zip(list(v.keys()), list(v.values())):
                    scores[j,index-1] += val
            scores = torch.Tensor(scores)
            sub_scores = scores.topk(20)[1]
            sub_scores = trans_to_cpu(sub_scores).detach().numpy()
            for score, target, mask in zip(sub_scores, targets, test_data.mask):
                hit.append(np.isin(target - 1, score))
                if len(np.where(score == target - 1)[0]) == 0:
                    mrr.append(0)
                else:
                    mrr.append(1 / (np.where(score == target - 1)[0][0] + 1))

        final_hit.append(np.mean(hit)*100)
        final_mrr.append(np.mean(mrr)*100)

        sr_gnn_hit.append(best_result[0])
        sr_gnn_mrr.append(best_result[1])

        iknn_hit.append(p_20)
        iknn_mrr.append(mrr_20)
    print()
    print('SR-GNN Score: ', np.mean(last_sr_gnn_hit), np.mean(last_sr_gnn_mrr))
    print('SR-GNN Last Score: ', np.mean(sr_gnn_hit), np.mean(sr_gnn_mrr))
    print('CKNN Score: ', np.mean(iknn_hit), np.mean(iknn_mrr))
    print('Proposed Method Score: ',np.mean(final_hit), np.mean(final_mrr))
    avg_p.append(np.mean(final_hit))
    avg_m.append(np.mean(final_mrr))
print('Method 3 Avg Proposed Method Score: ',np.mean(avg_p), np.mean(avg_m))

100%|██████████| 560/560 [00:00<00:00, 10248.85it/s]
/home/workplace/SR-GNN/final/model.py:18: UserWarning: Legacy tensor constructor is deprecated. Use: torch.tensor(...) for creating tensors from tensor-like objects; or torch.empty(...) for creating an uninitialized tensor with specific sizes. (Triggered internally at  /pytorch/torch/csrc/utils/tensor_new.cpp:478.)
  self.w_ih = Parameter(torch.Tensor(self.gate_size, self.input_size))
/usr/local/lib/python3.6/dist-packages/torch/optim/lr_scheduler.py:134: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
  0%|   

	Loss:	700.828


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	471.705


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	423.351


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	416.893


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	412.132


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	406.823


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	406.678


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	405.932


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	405.589


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	405.270


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	405.353


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	405.381


100%|██████████| 12/12 [00:00<00:00, 68.64it/s]

	Loss:	405.372
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/560 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

100%|██████████| 12/12 [00:00<00:00, 63.33it/s]


	Loss:	708.574


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	480.111


 58%|█████▊    | 7/12 [00:00<00:00, 62.16it/s]

	Loss:	435.597


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	429.174


100%|██████████| 12/12 [00:00<00:00, 62.78it/s]


	Loss:	423.871


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	419.774


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	418.317


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	418.274


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.026


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.912


100%|██████████| 12/12 [00:00<00:00, 61.03it/s]


	Loss:	418.112


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.282


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.177


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.535


100%|██████████| 12/12 [00:00<00:00, 65.06it/s]

	Loss:	417.301
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	714.918


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	487.374


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	436.865


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	429.061


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	423.245


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.465


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	416.988


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.487


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.086


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	414.977


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.320


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.130


100%|██████████| 12/12 [00:00<00:00, 72.93it/s]

	Loss:	414.971
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/563 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	711.933


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	487.344


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	436.876


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	428.655


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	422.220


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	416.661


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	416.071


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.728


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	414.676


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	414.619


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	414.850


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	414.576


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	414.370


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	414.511


100%|██████████| 12/12 [00:00<00:00, 69.28it/s]

	Loss:	414.536
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 50%|█████     | 6/12 [00:00<00:00, 50.70it/s]

	Loss:	705.520


 42%|████▏     | 5/12 [00:00<00:00, 48.46it/s]

	Loss:	487.486


 50%|█████     | 6/12 [00:00<00:00, 52.87it/s]

	Loss:	436.981


 50%|█████     | 6/12 [00:00<00:00, 53.45it/s]

	Loss:	428.902


 42%|████▏     | 5/12 [00:00<00:00, 47.84it/s]

	Loss:	422.429


 50%|█████     | 6/12 [00:00<00:00, 52.22it/s]

	Loss:	416.416


 42%|████▏     | 5/12 [00:00<00:00, 46.54it/s]

	Loss:	415.644


 50%|█████     | 6/12 [00:00<00:00, 50.52it/s]

	Loss:	415.079


 50%|█████     | 6/12 [00:00<00:00, 53.96it/s]

	Loss:	414.453


 50%|█████     | 6/12 [00:00<00:00, 54.37it/s]

	Loss:	414.424


 50%|█████     | 6/12 [00:00<00:00, 53.82it/s]

	Loss:	414.366


 50%|█████     | 6/12 [00:00<00:00, 53.50it/s]

	Loss:	414.280


 50%|█████     | 6/12 [00:00<00:00, 53.21it/s]

	Loss:	414.146


 50%|█████     | 6/12 [00:00<00:00, 53.22it/s]

	Loss:	414.155


100%|██████████| 12/12 [00:00<00:00, 53.81it/s]

-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 11%|█▏        | 521/4536 [00:00<00:00, 5207.86it/s]


SR-GNN Score:  49.338173340459726 39.97791667795731
SR-GNN Last Score:  49.48027060489741 39.98477209029336
CKNN Score:  40.071607630996915 24.28932932064422
Proposed Method Score:  54.88065544799984 40.69128085092071


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	704.347


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	473.784


100%|██████████| 12/12 [00:00<00:00, 61.80it/s]


	Loss:	423.883


100%|██████████| 12/12 [00:00<00:00, 63.08it/s]


	Loss:	416.342


100%|██████████| 12/12 [00:00<00:00, 62.98it/s]


	Loss:	410.454


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	404.883


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	404.167


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	403.778


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	403.266


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	403.181


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	403.213


100%|██████████| 12/12 [00:00<00:00, 63.90it/s]

	Loss:	403.171
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/560 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	701.729


100%|██████████| 12/12 [00:00<00:00, 62.99it/s]


	Loss:	484.184


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	434.611


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	427.280


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	420.657


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	415.696


100%|██████████| 12/12 [00:00<00:00, 62.20it/s]


	Loss:	414.475


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	413.999


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	413.177


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	413.126


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	413.079


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	412.961


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	413.583


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	413.403


100%|██████████| 12/12 [00:00<00:00, 65.71it/s]


	Loss:	414.035
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	725.777


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	490.029


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	444.380


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	435.715


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	429.348


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	424.273


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	423.999


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	423.594


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	422.075


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	422.568


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	422.517


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	422.414


100%|██████████| 12/12 [00:00<00:00, 72.15it/s]

	Loss:	422.309
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/563 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	709.606


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	487.980


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	440.117


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	431.401


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	425.370


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	419.503


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.760


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.957


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.198


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.199


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.209


100%|██████████| 12/12 [00:00<00:00, 68.85it/s]

	Loss:	417.413
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 50%|█████     | 6/12 [00:00<00:00, 51.63it/s]

	Loss:	710.799


 50%|█████     | 6/12 [00:00<00:00, 51.04it/s]

	Loss:	481.532


 50%|█████     | 6/12 [00:00<00:00, 52.65it/s]

	Loss:	434.903


 50%|█████     | 6/12 [00:00<00:00, 51.74it/s]

	Loss:	427.188


 50%|█████     | 6/12 [00:00<00:00, 51.02it/s]

	Loss:	422.039


 50%|█████     | 6/12 [00:00<00:00, 51.63it/s]

	Loss:	416.879


 50%|█████     | 6/12 [00:00<00:00, 52.44it/s]

	Loss:	416.153


 50%|█████     | 6/12 [00:00<00:00, 50.85it/s]

	Loss:	415.848


 42%|████▏     | 5/12 [00:00<00:00, 42.58it/s]

	Loss:	415.108


 42%|████▏     | 5/12 [00:00<00:00, 41.49it/s]

	Loss:	415.026


 50%|█████     | 6/12 [00:00<00:00, 51.08it/s]

	Loss:	415.114


 50%|█████     | 6/12 [00:00<00:00, 52.28it/s]

	Loss:	415.002


 50%|█████     | 6/12 [00:00<00:00, 52.91it/s]

	Loss:	415.109


 50%|█████     | 6/12 [00:00<00:00, 51.70it/s]

	Loss:	415.085


100%|██████████| 12/12 [00:00<00:00, 50.61it/s]

-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 33%|███▎      | 1486/4536 [00:00<00:00, 14851.83it/s]


SR-GNN Score:  48.82411896393794 39.76567653644829
SR-GNN Last Score:  48.89516759615678 39.77259106514005
CKNN Score:  40.07160763099692 24.289329320644217
Proposed Method Score:  54.5434932542232 40.48296437244213


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	704.759


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	468.802


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	424.316


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	417.137


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	411.625


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	406.368


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	406.087


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	405.494


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	404.919


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	405.074


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	404.831


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	404.793


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	404.982


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	404.973


100%|██████████| 12/12 [00:00<00:00, 63.74it/s]

	Loss:	404.792
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/560 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 50%|█████     | 6/12 [00:00<00:00, 49.71it/s]

	Loss:	702.288


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	480.339


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	435.304


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	425.467


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	418.479


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	413.652


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	412.838


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	411.125


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	412.161


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	411.391


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	411.321


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	411.440


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	411.793


100%|██████████| 12/12 [00:00<00:00, 65.44it/s]

	Loss:	410.737
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	715.255


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	490.357


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	440.372


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	432.227


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	427.070


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	421.240


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	420.465


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	419.700


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	420.140


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.778


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	419.302


100%|██████████| 12/12 [00:00<00:00, 72.51it/s]

	Loss:	418.755
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/563 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	722.716


100%|██████████| 12/12 [00:00<00:00, 62.21it/s]


	Loss:	484.888


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	436.830


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	429.558


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	424.769


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	419.140


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.878


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.096


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.669


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.343


100%|██████████| 12/12 [00:00<00:00, 71.32it/s]

	Loss:	417.741
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 50%|█████     | 6/12 [00:00<00:00, 51.44it/s]

	Loss:	704.568


 50%|█████     | 6/12 [00:00<00:00, 53.23it/s]

	Loss:	483.262


 50%|█████     | 6/12 [00:00<00:00, 53.35it/s]

	Loss:	433.484


 50%|█████     | 6/12 [00:00<00:00, 50.81it/s]

	Loss:	425.092


 50%|█████     | 6/12 [00:00<00:00, 53.31it/s]

	Loss:	419.098


 50%|█████     | 6/12 [00:00<00:00, 52.83it/s]

	Loss:	412.765


 50%|█████     | 6/12 [00:00<00:00, 50.26it/s]

	Loss:	411.915


 50%|█████     | 6/12 [00:00<00:00, 51.49it/s]

	Loss:	411.531


 50%|█████     | 6/12 [00:00<00:00, 51.04it/s]

	Loss:	410.963


 42%|████▏     | 5/12 [00:00<00:00, 49.76it/s]

	Loss:	410.838


 50%|█████     | 6/12 [00:00<00:00, 53.38it/s]

	Loss:	410.666


 50%|█████     | 6/12 [00:00<00:00, 52.55it/s]

	Loss:	410.732


100%|██████████| 12/12 [00:00<00:00, 53.49it/s]

-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 33%|███▎      | 1495/4536 [00:00<00:00, 14941.61it/s]


SR-GNN Score:  49.03070046783436 39.84645586608671
SR-GNN Last Score:  49.148988208462534 39.85774601779054
CKNN Score:  40.07160763099692 24.28932932064422
Proposed Method Score:  54.585128428000964 40.53196971328937


100%|██████████| 12/12 [00:00<00:00, 61.73it/s]


	Loss:	714.052


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	470.529


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	422.399


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	414.914


100%|██████████| 12/12 [00:00<00:00, 62.24it/s]


	Loss:	409.680


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	404.528


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	403.913


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	403.447


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	402.822


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	402.875


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	402.582


100%|██████████| 12/12 [00:00<00:00, 59.81it/s]

	Loss:	402.536



100%|██████████| 12/12 [00:00<00:00, 66.96it/s]

	Loss:	402.714
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/560 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

100%|██████████| 12/12 [00:00<00:00, 61.68it/s]


	Loss:	713.228


100%|██████████| 12/12 [00:00<00:00, 64.48it/s]


	Loss:	481.880


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	435.787


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	428.612


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	424.408


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	418.385


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.739


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.380


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	416.765


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.131


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.435


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	416.120


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	416.617


100%|██████████| 12/12 [00:00<00:00, 66.07it/s]

	Loss:	417.134
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	710.149


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	491.009


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	441.058


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	432.136


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	425.841


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	420.142


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	419.488


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.519


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.032


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.130


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.580


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.952


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.374


100%|██████████| 12/12 [00:00<00:00, 70.05it/s]

	Loss:	417.450
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/563 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	732.491


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	489.160


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	437.119


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	429.729


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	423.603


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.076


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.082


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	416.381


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.790


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.384


100%|██████████| 12/12 [00:00<00:00, 61.03it/s]


	Loss:	415.791


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.664


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.194


100%|██████████| 12/12 [00:00<00:00, 69.53it/s]

	Loss:	415.419
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 50%|█████     | 6/12 [00:00<00:00, 53.34it/s]

	Loss:	709.470


 42%|████▏     | 5/12 [00:00<00:00, 43.68it/s]

	Loss:	480.335


 42%|████▏     | 5/12 [00:00<00:00, 46.13it/s]

	Loss:	432.879


 50%|█████     | 6/12 [00:00<00:00, 50.87it/s]

	Loss:	425.341


 50%|█████     | 6/12 [00:00<00:00, 50.06it/s]

	Loss:	419.963


 42%|████▏     | 5/12 [00:00<00:00, 48.56it/s]

	Loss:	414.465


 42%|████▏     | 5/12 [00:00<00:00, 42.69it/s]

	Loss:	413.856


 50%|█████     | 6/12 [00:00<00:00, 51.79it/s]

	Loss:	413.339


 50%|█████     | 6/12 [00:00<00:00, 51.29it/s]

	Loss:	412.796


 50%|█████     | 6/12 [00:00<00:00, 52.86it/s]

	Loss:	412.858


 50%|█████     | 6/12 [00:00<00:00, 51.79it/s]

	Loss:	412.584


 50%|█████     | 6/12 [00:00<00:00, 52.01it/s]

	Loss:	412.725


 50%|█████     | 6/12 [00:00<00:00, 51.04it/s]

	Loss:	412.772


 50%|█████     | 6/12 [00:00<00:00, 51.15it/s]

	Loss:	412.624


 42%|████▏     | 5/12 [00:00<00:00, 45.71it/s]

	Loss:	412.626


 42%|████▏     | 5/12 [00:00<00:00, 48.88it/s]

	Loss:	412.797


100%|██████████| 12/12 [00:00<00:00, 53.34it/s]

-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 22%|██▏       | 1015/4536 [00:00<00:00, 10146.74it/s]


SR-GNN Score:  48.96572538920712 39.9082234640969
SR-GNN Last Score:  49.08122690896396 39.916720049274204
CKNN Score:  40.07160763099692 24.289329320644217
Proposed Method Score:  54.59708009181527 40.59958214407139


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	699.350


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	478.435


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	429.154


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	421.088


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	415.071


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	409.629


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	408.850


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	408.276


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	407.453


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	407.643


100%|██████████| 12/12 [00:00<00:00, 62.61it/s]


	Loss:	407.524


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	407.776


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	407.528


100%|██████████| 12/12 [00:00<00:00, 63.91it/s]


	Loss:	407.545
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/560 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

100%|██████████| 12/12 [00:00<00:00, 62.30it/s]


	Loss:	706.829


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	484.716


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	437.174


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	429.604


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	423.335


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.743


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	417.052


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	415.434


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	415.816


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	416.148


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	415.594


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	415.436


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	415.421


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	416.320


  0%|          | 0/92 [00:00<?, ?it/s]

	Loss:	415.581


100%|██████████| 12/12 [00:00<00:00, 63.99it/s]

	Loss:	415.732
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	709.936


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	489.791


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	439.069


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	431.135


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	424.506


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.509


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.433


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.576


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	416.325


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	416.493


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	416.564


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	415.813


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	416.341


100%|██████████| 12/12 [00:00<00:00, 71.53it/s]

	Loss:	416.089
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/563 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	716.870


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	487.051


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	439.109


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	430.953


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	425.356


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	419.868


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	419.368


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.607


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	418.246


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.487


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.918


100%|██████████| 12/12 [00:00<00:00, 64.05it/s]


	Loss:	418.003


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.792


  0%|          | 0/93 [00:00<?, ?it/s]

	Loss:	417.999


100%|██████████| 12/12 [00:00<00:00, 60.36it/s]

	Loss:	417.909



100%|██████████| 12/12 [00:00<00:00, 71.34it/s]

	Loss:	417.638
-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 50%|█████     | 6/12 [00:00<00:00, 51.97it/s]

	Loss:	747.887


 50%|█████     | 6/12 [00:00<00:00, 52.77it/s]

	Loss:	490.514


 42%|████▏     | 5/12 [00:00<00:00, 47.04it/s]

	Loss:	440.205


 42%|████▏     | 5/12 [00:00<00:00, 48.91it/s]

	Loss:	432.380


 50%|█████     | 6/12 [00:00<00:00, 51.84it/s]

	Loss:	426.371


 42%|████▏     | 5/12 [00:00<00:00, 49.97it/s]

	Loss:	420.777


 42%|████▏     | 5/12 [00:00<00:00, 49.68it/s]

	Loss:	420.184


 50%|█████     | 6/12 [00:00<00:00, 50.61it/s]

	Loss:	419.586


 50%|█████     | 6/12 [00:00<00:00, 50.97it/s]

	Loss:	418.985


 50%|█████     | 6/12 [00:00<00:00, 52.93it/s]

	Loss:	418.931


 50%|█████     | 6/12 [00:00<00:00, 53.22it/s]

	Loss:	419.028


 50%|█████     | 6/12 [00:00<00:00, 52.87it/s]

	Loss:	418.924


 50%|█████     | 6/12 [00:00<00:00, 52.95it/s]

	Loss:	418.766


 50%|█████     | 6/12 [00:00<00:00, 52.45it/s]

	Loss:	419.136


 42%|████▏     | 5/12 [00:00<00:00, 49.74it/s]

	Loss:	419.005


100%|██████████| 12/12 [00:00<00:00, 53.03it/s]

-------------------------------------------------------


  0%|          | 0/2260 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

  0%|          | 0/564 [00:00<?, ?it/s]

  0%|          | 0/12 [00:00<?, ?it/s]

 30%|███       | 1374/4536 [00:00<00:00, 13733.86it/s]


SR-GNN Score:  48.94107644493874 39.838279073837725
SR-GNN Last Score:  49.03347766074421 39.845204623013714
CKNN Score:  40.07160763099692 24.289329320644214
Proposed Method Score:  54.51918303846992 40.54599754062751
Method 1 Avg Proposed Method Score:  54.51918303846992 40.54599754062751


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	693.737


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	463.068


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	418.227


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	411.085


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	405.179


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	399.694


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	399.090


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	398.439


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	397.905


  0%|          | 0/91 [00:00<?, ?it/s]

	Loss:	397.664


 53%|█████▎    | 48/91 [00:02<00:02, 21.17it/s]


KeyboardInterrupt: 

In [7]:
print('done')

done


SR-GNN Score:  49.338173340459726 39.97791667795731
SR-GNN Last Score:  49.48027060489741 39.98477209029336
CKNN Score:  40.071607630996915 24.28932932064422
Proposed Method Score:  54.88065544799984 40.69128085092071
SR-GNN Score:  48.82411896393794 39.76567653644829
SR-GNN Last Score:  48.89516759615678 39.77259106514005
CKNN Score:  40.07160763099692 24.289329320644217
Proposed Method Score:  54.5434932542232 40.48296437244213
SR-GNN Score:  49.03070046783436 39.84645586608671
SR-GNN Last Score:  49.148988208462534 39.85774601779054
CKNN Score:  40.07160763099692 24.28932932064422
Proposed Method Score:  54.585128428000964 40.53196971328937
SR-GNN Score:  48.96572538920712 39.9082234640969
SR-GNN Last Score:  49.08122690896396 39.916720049274204
CKNN Score:  40.07160763099692 24.289329320644217
Proposed Method Score:  54.59708009181527 40.59958214407139
SR-GNN Score:  48.94107644493874 39.838279073837725
SR-GNN Last Score:  49.03347766074421 39.845204623013714
CKNN Score:  40.07160763099692 24.289329320644214
Proposed Method Score:  54.51918303846992 40.54599754062751
Method 1 Avg Proposed Method Score:  54.51918303846992 40.54599754062751

SR-GNN Score:  48.11847103655715 39.2057053415415
SR-GNN Last Score:  48.15412166044307 39.20580884513826
CKNN Score:  44.816038164553774 32.47939661724543
Proposed Method Score:  53.939598209922906 41.098498802422924
SR-GNN Score:  47.728084625901445 38.848617854854226
SR-GNN Last Score:  47.745909937844395 38.84875388815281
CKNN Score:  44.81603816455377 32.47939661724543
Proposed Method Score:  53.708689658656866 41.038768206549165
SR-GNN Score:  48.24889650150303 38.88524991286906
SR-GNN Last Score:  48.26078004279834 38.885435547942976
CKNN Score:  44.81603816455377 32.47939661724543
Proposed Method Score:  53.89202085875224 40.73011427902729
SR-GNN Score:  48.49108362989154 39.02498285116509
SR-GNN Last Score:  48.53545761440087 39.02659248494556
CKNN Score:  44.816038164553774 32.47939661724543
Proposed Method Score:  53.921393354940996 40.63162025548813
SR-GNN Score:  48.72189526857687 39.050454661048
SR-GNN Last Score:  48.757394456184336 39.05186666855066
CKNN Score:  44.81603816455377 32.47939661724543
Proposed Method Score:  53.98861270420677 40.54752250153825
Method 1 Avg Proposed Method Score:  53.98861270420677 40.54752250153825

SR-GNN Score:  48.52702499350679 39.106945040227394
SR-GNN Last Score:  48.56841752034572 39.112912183746786
CKNN Score:  44.81603816455377 32.47939661724543
Proposed Method Score:  53.98045400718165 40.60424014995526
SR-GNN Score:  48.63624919077486 39.212073338877154
SR-GNN Last Score:  48.671728499493945 39.21761018799983
CKNN Score:  44.816038164553774 32.47939661724543
Proposed Method Score:  54.08103659388731 40.60352193671038
SR-GNN Score:  48.55870265102725 39.19346697201373
SR-GNN Last Score:  48.59417182491751 39.198353745237206
CKNN Score:  44.81603816455377 32.47939661724543
Proposed Method Score:  54.063356711686176 40.61730094980477
SR-GNN Score:  48.533611894126516 39.12721319877099
SR-GNN Last Score:  48.58091451160068 39.13155699719186
CKNN Score:  44.81603816455377 32.47939661724543
Proposed Method Score:  54.04176053849841 40.545403776459835
SR-GNN Score:  48.58447411683145 39.1367224817727
SR-GNN Last Score:  48.63766594158474 39.14133812332717
CKNN Score:  44.81603816455377 32.47939661724543
Proposed Method Score:  54.063434407225905 40.54142512509033
Method 2 Avg Proposed Method Score:  54.063434407225905 40.54142512509033

SR-GNN Score:  48.590765339984316 39.12659448086034
SR-GNN Last Score:  48.642339565222635 39.130790518637134
CKNN Score:  44.81603816455376 32.47939661724543
Proposed Method Score:  54.05216522092618 40.52465604434711
SR-GNN Score:  48.5604520681143 39.14678786988186
SR-GNN Last Score:  48.61068352399174 39.15063423784391
CKNN Score:  44.81603816455377 32.47939661724543
Proposed Method Score:  54.05469440346543 40.53582844787609
SR-GNN Score:  48.567676866887645 39.13613415565544
SR-GNN Last Score:  48.61677213330588 39.13968464915888
CKNN Score:  44.81603816455376 32.47939661724542
Proposed Method Score:  54.10325061945461 40.541453194420896
SR-GNN Score:  48.48246242571709 39.17214756440152
SR-GNN Last Score:  48.5331257414891 39.17544673972324
CKNN Score:  44.81603816455376 32.47939661724543
Proposed Method Score:  54.07893248166918 40.59627225120298
SR-GNN Score:  48.48429417220811 39.18395485627738
SR-GNN Last Score:  48.53394399978918 39.187034086577654
CKNN Score:  44.81603816455377 32.47939661724543
Proposed Method Score:  54.06491956683783 40.58811045570778
Method 3 Avg Proposed Method Score:  54.06491956683783 40.58811045570778

40k
CKNN Score:  44.81603816455377 32.47939661724543

SR-GNN Last Score:    48.29073274233419 39.003691486946046
Method 1 Avg Proposed Method Score:  53.98861270420677 40.54752250153825

SR-GNN Last Score:   48.610579659588524 39.16035424750057
Method 2 Avg Proposed Method Score:  54.063434407225905 40.54142512509033

SR-GNN Last Score:   48.58737299275971 39.156718046388164
Method 3 Avg Proposed Method Score:  54.06491956683783 40.58811045570778

2018
CKNN Score:  8.965137202665257 5.687222376937899

SR-GNN Last Score:  24.12804519933014 14.705060071171676
Method 1 Avg Proposed Method Score:  24.513084956296794 12.923072506624601

SR-GNN Last Score:  24.30988855051716 14.588509508467718
Method 2 Avg Proposed Method Score:  24.67480070201331 12.747575408266078

SR-GNN Last Score:  24.111452628616544 14.477994448393519
Method 3 Avg Proposed Method Score:  24.54353790525956 12.757424840740866

40k
data: 40000
user: 9110
item: 21874
avg_item: 4.390779363336993

2018
data: 2467321
user: 1361089
item: 492767
avg_item: 1.8127550806743717